### PR TITLE
chore: update api-client to surface delinquency warnings

### PR DESCRIPTION
**Pull request by zwhitfield3**
## Description

This PR updates the `api-client.ts` to surface any delinquency warnings around accessing `/account` or `/team` resources.

### CODE REVIEW

File: src/api-client.ts [hunk 1]
```diff
@@ -117,7 +143,41 @@ export class APIClient {
...
+              ])
+              response = originalRequest
+              delinquencyResponse = accountDelinquency
+            }
+          } else {
+            response = await super.request<T>(url, opts)
+          }
+
+          if (delinquencyResponse && delinquencyInfo.teamId) {
+            warn('This team is delinquent and will be suspended in ${delinquencyResponse.scheduled_suspension_time}')
```
Comment (SandyPantsLai) [inline src/api-client.ts:174]: This team is delinquent with payment and we will suspend it on ${delinquencyResponse.scheduled_suspension_time}.

File: src/api-client.ts [hunk 2]
```diff
@@ -117,7 +143,41 @@ export class APIClient {
...
+          }
+
+          if (delinquencyResponse && !delinquencyInfo.teamId) {
+            warn('This account is delinquent and will be suspended in ${delinquencyResponse.scheduled_suspension_time}')
```
Comment (SandyPantsLai) [inline src/api-client.ts:178]: This account is delinquent with payment and we will suspend it on ${delinquencyResponse.scheduled_suspension_time}.

**Comment by zwhitfield3** [comment]
@mattgraham - requesting a design review.

**Comment by zwhitfield3** [comment]
Closed in favor of another solution.

**Pull request closed without merging**